pkg/db/db.go

```diff
@@ -0,0 +1,93 @@
+package db
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/types"
+)
+
+type Database struct {
+	tableName string
+	pool      *pgxpool.Pool
+}
+
+func NewClient(ctx context.Context, connString, providerTableName string) (*Database, error) {
```

Comment:
Don't assume the client only work on a single table, accept a map and validate that the input (map) contains a name mapping for all the tables you need in order to make the client work 
```suggestion
func NewClient(ctx context.Context, connString, tables map[string]string) (*Database, error) {
    if _, ok := tables["providers"]; ! ok {
        return fmt.Errorf("Incomplete table name mapping, missing a table name for \"providers\" table")
    }

```